samtranslator/utils/utils.py

```diff
@@ -71,3 +71,8 @@ def dict_deep_set(d: Any, path: str, value: Any) -> None:
     if not isinstance(d, dict):
         raise InvalidValueType(relative_path)
     d[_path_nodes[0]] = value
+
+
+def remove_none_items(d: Dict[Any, Any]) -> Dict[Any, Any]:
```

Comment:
nit: "items" is ambiguous; "value" would be clearer

Also please move this under `internal` so as to not increase public surface.